packages/rpcs/wallet_sendTransaction/index.js

```diff
@@ -0,0 +1,230 @@
+
+      throw err
+    }
+
+    if (
+      network.type === 'cfx' &&
+      app.currentNetwork.name === CFX_MAINNET_NAME
+    ) {
+      try {
+        const gasPrice = await cfx_gasPrice({errorFallThrough: true}, [])
+        if (
+          BigNumber.from(gasPrice).gt(
+            BigNumber.from(params[0].gasPrice || '0x0'),
+          )
+        ) {
+          params[0].gasPrice = gasPrice
+        }
+      } catch (err) {} // eslint-disable-line no-empty
+    }
+
+    return await wallet_addPendingUserAuthRequest({
+      appId: app.eid,
+      req: {method: NAME, params},
+    })
+  }
+
+  const authReqId = params?.authReqId
+  let authReq
+  if (authReqId) {
+    authReq = getAuthReqById(authReqId)
+    if (!authReq) throw InvalidParams(`Invalid authReqId ${authReqId}`)
+    if (authReq.processed)
+      throw InvalidParams(`Already processing auth req ${authReqId}`)
+    t({eid: authReqId, authReq: {processed: true}})
+  }
+
+  // tx array [tx]
+  const tx = params.authReqId ? params.tx : params
+  let addr = findAddress({
+    // filter by app.currentNetwork and app.currentAccount
+    appId: authReq?.app?.eid,
+    // filter by current network
+    networkId: !authReqId && network.eid,
+    value: tx[0].from,
+  })
+  addr = addr[0] || addr
+  if (!addr) throw InvalidParams(`Invalid from address ${tx[0].from}`)
+
+  let signed
+  try {
+    signed = await signTxFn(
+      {
+        network: authReqId ? authReq.app.currentNetwork : network,
+        errorFallThrough: true,
+      },
+      tx.concat({
+        returnTxMeta: true,
+      }),
+    )
+  } catch (err) {
+    if (authReqId) await wallet_userRejectedAuthRequest({authReqId})
+    throw err
+  }
+
+  if (!signed) {
+    if (authReqId) await wallet_userRejectedAuthRequest({authReqId})
+    throw Server(`Server error while signning tx`)
+  }
+  const {raw: rawtx, txMeta} = signed
+
+  const txhash = getTxHashFromRawTx(rawtx)
+  const duptx = getAddrTxByHash({addressId: addr, txhash})
+
+  if (duptx) throw InvalidParams('duplicate tx')
+
+  const blockNumber =
+    network.type === 'eth' &&
+    (await eth_blockNumber({errorFallThrough: true}, []))
+
+  const dbtxs = [
+    {eid: 'newTxPayload', txPayload: txMeta},
+    {eid: 'newTxExtra', txExtra: {ok: false}},
+    {
+      eid: 'newTxId',
+      tx: {
+        fromFluent: true,
+        txPayload: 'newTxPayload',
+        hash: txhash,
+        raw: rawtx,
+        status: 0,
+        created: new Date().getTime(),
+        txExtra: 'newTxExtra',
+      },
+    },
+    blockNumber && {eid: 'newTxId', tx: {blockNumber}},
+    {eid: addr, address: {tx: 'newTxId'}},
+    authReqId && {eid: authReq.app.eid, app: {tx: 'newTxId'}},
+  ]
+
+  const {
+    tempids: {newTxId},
+  } = t(dbtxs)
+
+  try {
+    enrichTxFn(
+      {
+        errorFallThrough: true,
+        network: authReqId ? authReq.app.currentNetwork : network,
+      },
+      {txhash},
+    )
+  } catch (err) {} // eslint-disable-line no-empty
```

Comment:
I think we should handle the error or print it or add it to sentry, otherwise, we will not know the error information if the error happens

Reply:
Unexpected errors are handled at https://github.com/Conflux-Chain/helios/blob/d22c3e6831f7f60024fca199149aad168613b7cb/packages/rpc-engine/src/error.js#L22

these are expected error